src/main/scala/it/pagopa/interop/agreementprocess/service/impl/QueueServiceImpl.scala

```diff
@@ -0,0 +1,14 @@
+package it.pagopa.interop.agreementprocess.service.impl
+
+import it.pagopa.interop.agreementprocess.service.QueueService
+import it.pagopa.interop.commons.queue.impl.SQSHandler
+import spray.json.JsonWriter
+
+import scala.concurrent.{ExecutionContextExecutor, Future}
+
+final class QueueServiceImpl(queueUrl: String)(implicit blockingEc: ExecutionContextExecutor) extends QueueService {
+
+  val sqsHandler: SQSHandler = SQSHandler(queueUrl)(blockingEc)
+
+  override def send[T: JsonWriter](message: T): Future[String] = sqsHandler.send(message)
+}
```

Comment:
I'm quite against this OOP tradition of wrapping stuff in classes and using traits when there are already some others to leverage.

If the only method exposed is `send` which is the purpose of `SQSHandler` extending [QueueWriter](https://github.com/pagopa/interop-commons/blob/1.0.x/queue-manager/src/main/scala/it/pagopa/interop/commons/queue/QueueWriter.scala#L9)? Can't we decouple, even for tests, relying on `QueueWriter` instead of `QueueService`?

Reply:
The interface is not exactly the same, but I want to encourage the discussion. Won't it be useful to modify the trait defining `def send[T: JsonWriter](message: T)` directly in commons?

Reply:
I appreciate this point.
Currently, `QueueWriter` is defined to work on the Message type, strictly correlated to Akka event structure, so none of its functions could be used.
`SQSHandler` is a class and not a trait.
I would move functions `def send[T: JsonWriter](message: T): Future[String]` and  `def deleteMessage(receiptHandle: String): Future[Unit]` from SQSHandler to QueueWriter, but I need to check where these functions are used.